py/objarray.c

```diff
@@ -258,6 +258,19 @@ STATIC mp_obj_t array_unary_op(mp_unary_op_t op, mp_obj_t o_in) {
     }
 }
 
+STATIC int typecode_for_comparison(int typecode, bool ignore_sign) {
+    if (typecode == BYTEARRAY_TYPECODE) {
+        return 'B';
+    }
+    if (typecode == 'f' || typecode == 'd') {
+        return typecode;
+    }
```

Comment:
I think you can just remove this if statement completely, it won't be caught by the if below and should be equivalent.  My testing shows that saves a handful of bytes.

Reply:
Without this the typecode always becomes uppercase, so the `lhs_code != 'f' && lhs_code != 'd'` below are never true and we'd be comparing floating point arrays. I can go back to my first attempt though, which removes this bit and below compares `lhs_code != 'F' && lhs_code != 'D'`. Works, but is just somewhat confusing because `F` and `D` aren't valid typecodes anywhere. But I'd add a comment to explain it.

Reply:
Changed this, added a comment, and removed the duplicate comparison against 'Z' to save extra bytes. For unix this is now +96 instead of +144, of which +64 is for the first commit.

Reply:
Nice little improvement!

Could the function always return lowercase instead of uppercase?  Then it's more obvious with the f/d.

Reply:
Yes but that needs removing the early return in case of BYTEARRAY_TYPECODE and using `is_unsigned` instead to be able to keep the same amount/type of instructions. Not really a problem, will do so right away.